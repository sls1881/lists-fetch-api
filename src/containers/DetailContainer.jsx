import React, { Component } from 'react';
import CharacterDetailItem from '../components/details/CharacterDetailItem';
import { getCharacter } from '../services/RickAndMortyApi';

export default class DetailContainer extends Component {
  state = {
    character: {},
  };

  async componentDidMount() {
    const { id } = this.props.match.params;
    const character = await getCharacter(id);
    this.setState({
      character,
    });
  }

  render() {
    const { character } = this.state;
    return (
      <div>
        <h1>Details Page</h1>
        <CharacterDetailItem character={character} />
      </div>
    );
  }
}
