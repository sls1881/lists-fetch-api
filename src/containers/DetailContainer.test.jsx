import React from 'react';
import { render, screen } from '@testing-library/react';
import DetailContainer from './DetailContainer';

describe('Rick and Morty API', () => {
  it('should render Rick and Morty character data', async () => {
    render(<DetailContainer match={{ params: { character: '' } }} />);

    screen.getByText('Details Page');

    const ul = await screen.findByRole('list', { name: 'character' });

    expect(ul).toMatchSnapshot();
  });
});
