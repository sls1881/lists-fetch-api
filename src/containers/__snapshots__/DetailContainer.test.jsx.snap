// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rick and Morty API should render Rick and Morty character data 1`] = `
<ul
  aria-label="character"
>
  <li>
    <h2 />
    <p />
    <p />
    <p />
    <img />
  </li>
</ul>
`;
